### DISCUSSION TITLE
Lit web component in grapesjs - adoptedStyleSheets limitation

### DISCUSSION THREAD

**Discussion by R-Tellier** [GrapesJS/grapesjs#4325] [Q&A]
Hello,
I'm trying to put some Lit web-components into grapesjs.

Passing a web-component with adoptedStyleSheets (lit behavior) from one document to another does not work.

So, I was wondering why the element is created inside the main document (certainly done by backbone) and then moved into the "canvas" iframe, do you know if this behavior can be changed and see some downside to create elements directly in iframe?

**Comment by artf** [comment]
Did you check these issues already?
https://github.com/artf/grapesjs/issues/3674
https://github.com/artf/grapesjs/issues/3693

**Reply by R-Tellier** [reply]
Yes we see the work around and we have a working one.
But this lead to some dirty code and were wondering why this is done this way.

In other words, is there an advantage to create element on the main document and then move it to the iframe?
